src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -181,18 +192,22 @@ impl TimeTrigger {
             let now: std::time::Duration = SystemTime::now()
                 .duration_since(UNIX_EPOCH)
                 .expect("system time before Unix epoch");
-            NaiveDateTime::from_timestamp_opt(now.as_secs() as i64, now.subsec_nanos())
+            DateTime::from_timestamp(now.as_secs() as i64, now.subsec_nanos())
                 .unwrap()
+                .naive_local()
                 .and_local_timezone(Local)
                 .unwrap()
         };
 
         #[cfg(not(test))]
         let current = Local::now();
-        let next_time = TimeTrigger::get_next_time(current, config.interval, config.modulate);
+        let next_time =
+            TimeTrigger::get_next_time(current, config.interval, config.modulate).unwrap();
```

Comment:
need to remove this unwrap and handle the error

Reply:
Added a match which panics on error. This can be handled more gracefully with changes from #347 which changes where `get_next_time` is called to a method that returns a result.

Reply:
Swapped to `unwrap_or` with an i64 max similar to the next_roll_time attempt with rand.

Reply:
Updated new to return a Result and passed the result up